app/src/main/java/org/zalando/nakadi/config/WebConfig.java

```diff
@@ -83,35 +83,40 @@ public FilterRegistrationBean traceRequestFilter() {
     }
 
     @Bean
-    public FilterRegistrationBean monitoringRequestFilter() {
+    public FilterRegistrationBean flowIdRequestFilter() {
         return createFilterRegistrationBean(
-                new MonitoringRequestFilter(metricRegistry, perPathMetricRegistry, authorizationService),
+                new FlowIdRequestFilter(),
                 Ordered.HIGHEST_PRECEDENCE + 10);
     }
 
     @Bean
-    public FilterRegistrationBean flowIdRequestFilter() {
+    public FilterRegistrationBean requestRejectedFilter() {
         return createFilterRegistrationBean(
-                new FlowIdRequestFilter(),
+                new RequestRejectedFilter(),
```

Comment:
the logging filter will not be called, in the previous version it was called. it could be important to log rejected requests

Reply:
I meant the `LoggingFilter` is called after `RequestRejectedFilter`

Reply:
Yeah, it is tricky.  We might have to split some of the filters to pre- and post- client id phase somehow.

Reply:
Changed precedence of this filter to make sure rejected requests are logged as well.
This starts to look like too much magic, though.

Reply:
And now it fails `InvalidRequestAT` due to unexpected error `500`, so handling of rejected exception is broken by this.

Looks like we will have to touch the plugin code as well to change precedence of filters there...

Ideally we would express dependencies and build a directed graph to determine the order of filters instead of messing with explicit ordering.